src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -821,11 +821,11 @@ internal ResultVersionOne CreateResultVersionOne(Result v2Result)
             return result;
         }
 
-        internal static RuleVersionOne CreateRuleVersionOne(IRule v2IRule)
+        internal static RuleVersionOne CreateRuleVersionOne(MessageDescriptor v2IRule)
         {
             RuleVersionOne rule = null;
 
-            var properties = v2IRule is Rule v2Rule ? v2Rule.Properties : null;
+            var properties = v2IRule.Properties;
```

Comment:
>var properties = v2IRule.Properties [](start = 12, length = 35)

No need for the temporary; just assign this to `RuleVersionOne.Properties` on L839.